packages/google_maps_flutter/google_maps_flutter_web/lib/src/markers.dart

```diff
@@ -0,0 +1,146 @@
+    @required StreamController<MapEvent> stream,
+  })  : _streamController = stream,
+        _markerIdToController = Map<MarkerId, MarkerController>();
+
+  /// Returns the cache of markers. Test only.
+  @visibleForTesting
+  Map<MarkerId, MarkerController> get markers => _markerIdToController;
+
+  /// Adds a set of [Marker] objects to the cache.
+  ///
+  /// (Wraps each Marker into its corresponding [MarkerController])
+  void addMarkers(Set<Marker> markersToAdd) {
+    if (markersToAdd != null) {
+      markersToAdd.forEach(_addMarker);
+    }
+  }
+
+  void _addMarker(Marker marker) {
+    if (marker == null) return;
+
+    final infoWindowOptions = _infoWindowOptionsFromMarker(marker);
+    gmaps.InfoWindow gmInfoWindow;
+
+    if (infoWindowOptions != null) {
+      gmInfoWindow = gmaps.InfoWindow(infoWindowOptions)
+        ..addListener('click', () {
+          _onInfoWindowTap(marker.markerId);
+        });
+    }
+
+    final currentMarker = _markerIdToController[marker.markerId]?.marker;
+
+    final populationOptions = _markerOptionsFromMarker(marker, currentMarker);
+    gmaps.Marker gmMarker = gmaps.Marker(populationOptions);
+    gmMarker.map = googleMap;
+    MarkerController controller = MarkerController(
+      marker: gmMarker,
+      infoWindow: gmInfoWindow,
+      consumeTapEvents: marker.consumeTapEvents,
+      onTap: () {
+        // TODO: If has infowindow...
+        this.showMarkerInfoWindow(marker.markerId);
+        _onMarkerTap(marker.markerId);
+      },
+      onDragEnd: (gmaps.LatLng latLng) {
+        _onMarkerDragEnd(marker.markerId, latLng);
+      },
+    );
+    _markerIdToController[marker.markerId] = controller;
+  }
+
+  /// Updates a set of [Marker] objects with new options.
+  void changeMarkers(Set<Marker> markersToChange) {
+    if (markersToChange != null) {
+      markersToChange.forEach(_changeMarker);
+    }
+  }
+
+  void _changeMarker(Marker marker) {
+    if (marker == null) {
+      return;
+    }
+    MarkerController markerController = _markerIdToController[marker.markerId];
+    if (markerController != null) {
+      final markerOptions = _markerOptionsFromMarker(
+        marker,
+        markerController.marker,
+      );
+      final infoWindow = _infoWindowOptionsFromMarker(marker);
+      markerController.update(
+        markerOptions,
+        newInfoWindowContent: infoWindow?.content,
+      );
+    }
+  }
+
+  /// Removes a set of [MarkerId]s from the cache.
+  void removeMarkers(Set<MarkerId> markerIdsToRemove) {
+    if (markerIdsToRemove == null) {
+      return;
+    }
+    markerIdsToRemove.forEach(_removeMarker);
+  }
+
+  void _removeMarker(MarkerId markerId) {
+    if (markerId == null) return;
+
+    final MarkerController markerController = _markerIdToController[markerId];
+    if (markerController != null) {
+      markerController.remove();
+      _markerIdToController.remove(markerId);
+    }
+  }
+
+  // InfoWindow...
+
+  /// Shows the [InfoWindow] of a Marker.
```

Comment:
Maybe add See also to point to the [hideMarkerInfoWindow], and [isInfoWindowShown]

Reply:
Added!